server/app/assets/javascripts/admin_predicate_configuration.ts

```diff
@@ -52,6 +59,69 @@ class AdminPredicateConfiguration {
     )
   }
 
+  validateSubmit(event: Event) {
+    let hasSelectionMissing = false
+    let hasValueMissing = false
+
+    // Check if any scalars are missing a value.
+    Array.from(
+      document.querySelectorAll<HTMLSelectElement>('.cf-scalar-select select'),
+    ).forEach((el: HTMLSelectElement) => {
+      if (el.options[el.options.selectedIndex].value == '') {
+        hasSelectionMissing = true
+      }
+    })
+
+    // Check if any operators are missing a value.
+    Array.from(
+      document.querySelectorAll<HTMLSelectElement>(
+        '.cf-operator-select select',
+      ),
+    ).forEach((el: HTMLSelectElement) => {
+      if (el.options[el.options.selectedIndex].value == '') {
+        hasSelectionMissing = true
+      }
+    })
+
+    // Check if any inputs are missing a value.
+    document
+      ?.querySelector('#predicate-config-value-row-container')
+      ?.querySelectorAll('input')
+      .forEach((input) => {
+        if (input.value == '') {
+          hasValueMissing = true
+        }
+      })
+
+    // If there are issues with any of the fields, we show a toast and prevent submit.
+    if (hasValueMissing || hasSelectionMissing) {
+      event.preventDefault()
+      // Scroll to the top of the page to ensure the user sees the error message.
+      window.scrollTo(0, 0)
+      let errorMessage
+      if (hasSelectionMissing && hasValueMissing) {
+        errorMessage =
+          'One or more form fields or drop-down is missing an entry. Please fill out all form fields and dropdowns before saving.'
```

Comment:
nit: here we use "drop-down" and later "dropdowns". I think this one should be plural and we should be consistent between the two with dash or no dash.